src/components/core/geoChart/GeoChartRenderer.tsx

```diff
@@ -58,6 +68,25 @@ export default class GeoChartRenderer extends React.Component<IGeoChartRendererP
         mapboxgl.accessToken = props.config.mapboxToken;
     }
 
+    public shouldComponentUpdate(nextProps: IGeoChartRendererProps) {
```

Comment:
For execution/bucket change, we already handle it in [GeoValidatorHOC](https://github.com/gooddata/gooddata-react-components/blob/develop/src/components/core/base/GeoValidatorHOC.tsx#L44).
In case execution/bucket changes, Geo chart will be unmounted by [BaseVisualization](https://github.com/gooddata/gooddata-react-components/blob/develop/src/components/core/base/BaseVisualization.tsx#L35) for showing loading icon, after that, geo chart is rendered freshly.

Reply:
ok